### DISCUSSION TITLE
No Issue, just asking!

### DISCUSSION THREAD

**Discussion by snzyeee** [omerfaruk-aran/esphome_samsung_hvac_bus#163] [Questions] [answered]
Hey,

im very interested in trying this integration. What i can see so far looks amazing. 
What iam especially looking for:

Is it possible to control the maximum power the outdoor unit is able to get?

**Comment by lanwin** [comment]
Hi @snzyeee, if you want to ask question, please use the "Discussions" section.

I am not sure what you mean. But it is possible to get the power the device currently uses.

**Reply by snzyeee** [reply]
Iam sorry for my mistake of not knowing how to ask a proper question.

The thing I want to do is limit the maximum power, so the outside unit can consume 500w max.

**Reply by lanwin** [reply]
Sorry, as far as I know, this is not possible.

**Reply by brazoayeye** [reply]
For my heat pump that's not possible (from what I know).
On the other side I can regulate heating setpoint and the heatpumpt modulate when it's close to it so I made a pid to match the photocoltaic production.
That's not possibile for DWH, my heat pump always runs at 3kW and it's not possible to reduce it changing the setpoint

**Reply by snzyeee** [reply]
> For my heat pump that's not possible (from what I know). On the other side I can regulate heating setpoint and the heatpumpt modulate when it's close to it so I made a pid to match the photocoltaic production. That's not possibile for DWH, my heat pump always runs at 3kW and it's not possible to reduce it changing the setpoint

Interesting! Thank you!

**Comment by tfoutfou** [comment]
> Hey,
> 
> im very interested in trying this integration. What i can see so far looks amazing. What iam especially looking for:
> 
> Is it possible to control the maximum power the outdoor unit is able to get?

i don't think you can limit the current with any official wired remote or official application at all , this is not a feature that exist !

The HVAC system will regulate itself to provide the temperature you asked in your room

it can go as low as ~500 W and up to 4600W at full power   ( i own a 14Kw/48k BTU cooling capacity unit)



i am thinking of ways you can cheat the system , maybe by replacing the wired remote temperature sensor with a digital potentiometer controled with an arduino , and set the same temperature as the "fake" sensor on the wired remote , this way the HVAC system will think you are already at the desired temperature and will run very slow with the minimal energy possible !

**Comment by projectcobalt** [comment]
Is your outdoor unit sending data on message 0x80A7? If so, you will have a DRM enabled outdoor unit of which will have some terminals on the main PCB that when bridged allows you to limit the outdoor units electrical consumption in a few different modes. 

Mine for example limits the unit to 75%, 50% of consumption with final mode being compressor off. The percentages are based on the outdoor units maximum electrical consumption..

There isn't a way to send the unit a specific value to hold for electrical input or compressor frequency.

**Reply by snzyeee** [reply]
This sounds very interesting. I will check for the data sent. Even limit it to 50% would be very great. 

But that would mean, if I want to change the value, I need to remove the bridge? I'm curious, because my outdoor unit is placed 8m high on a wall. So I'm not able to get there often. 

Still this would be a good workaround for autumn - winter. 

Is there any documentation I can read about what to bridge? Or can you provide that information?

**Reply by snzyeee** [reply]
So i just installed the "Thing" and there is indeed information sent on 0x80A7. 

What is the next step? Can you please provide some information about what to bridge?

Kind Regards!

**Reply by snzyeee** [reply]
Am i on the right foot here? Do i need to bridge one of the DRM Ports with "C" on the lower part of the diagram?

![image](https://github.com/user-attachments/assets/c5c50f08-953f-4754-b86c-9bc013f156af)

**Reply by projectcobalt** [reply, marked as answer]
Hey Mate, yeah that's the one C (Common) to 3/2 or 1... From memory DRM-3 limits outdoor unit power consumption to 75% of rated capacity, DRM-2 is a 50% limit and DRM-1 is Compressor off... Pretty sure its 5v logic so can be driven by relay or for testing just bridge it...

Add 0x8236, 0x8237 & 0x8238 as custom messages, these are the frequency variables of the compressor.. Get the unit running at maximum Heat or Cool (Depending where you are) then bridge one.. Within a few seconds those variables will change and also 0x80A7 will give you a number..

There is also 12v on CN303, so you should be able to power an esp32 relay board and let it live inside the outdoor unit.

**Reply by snzyeee** [reply]
Thank you a lot for that Information. I will keep an update here. You just made my day!

**Reply by snzyeee** [reply]
Well, today was the day of testing. Sadly i wasnt successful. 

I installed a Relay wich was clearly working, but there was no effect. 

Maybe it needs to be activated somehow or is it a "country-thing"?
